src/lib/viewers/doc/SSRRenderer.js

```diff
@@ -0,0 +1,169 @@
+        this.cache = new CellMeasurerCache({
+            defaultHeight: getDimensionsFromRep(this.currentRep),
+            fixedWidth: true
+        });
+
+        this.resizePage = this.resizePage.bind(this);
+    }
+
+    /**
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        if (this.listComponent) {
+            unmountComponentAtNode(this.docEl);
+            this.docComponent = null;
+        }
+    }
+
+    /* eslint-disable react/prop-types */
+    /**
+     * Renders cell
+     *
+     * @private
+     * @param {Object} pageInfo - Page data
+     * @param {number} pageInfo.columnIndex - Page column index
+     * @param {string} pageInfo.key - Page key
+     * @param {number} pageInfo.rowIndex - Page row index
+     * @param {string} cellInfo.style - Page style
+     * @return {Function} Page renderer function
+     */
+    pageRenderer = ({ index, key, style }) => {
+        const { url_template: template } = this.currentRep.content;
+        let url = this.createImageSource(template, this.asset);
+        url = url.replace(this.asset, `${index + 1}.png`);
+
+        return (
+            <CellMeasurer cache={this.cache} columnIndex={0} key={key} parent={parent} rowIndex={index}>
+                {({ measure }) => {
+                    return (
+                        <div style={style} className={'bp-page'} key={key} data-page-number={index + 1}>
+                            <img
+                                className={'bp-hidden'}
+                                onLoad={() => {
+                                    this.resizePage(measure, index + 1);
+                                }}
+                                alt={`page ${index + 1}`}
+                                src={url}
+                                validate={'always'}
+                            />
+                        </div>
+                    );
+                }}
+            </CellMeasurer>
+        );
+    };
+    /* eslint-enable react/prop-types */
+
+    /**
+     * Resizes a page based on representation and scale value
+     *
+     * @private
+     * @param {Function} measure - The measure function provided by the CellMeasurer component
+     * @param {number} pageNumber - The page number to resize
+     * @return {void}
+     */
+    resizePage(measure, pageNumber) {
+        if (this.currentScaleValue !== 'auto') {
+            return;
+        }
+
+        const page = this.docEl.querySelector(`[data-page-number="${pageNumber}"]`);
+        const image = page.firstChild;
+
+        const maxScale = MAX_AUTO_SCALE / (getDimensionsFromRep(this.currentRep) / MINIMUM_REP_DIMENSION);
+        const scaleValue = Math.min((this.docEl.clientWidth - ROW_HEIGHT_PADDING) / image.naturalWidth, maxScale);
+
+        image.width = image.naturalWidth * scaleValue;
+        image.height = image.naturalHeight * scaleValue;
+
+        this.cache.clear(pageNumber - 1);
+        measure();
```

Comment:
This is a method provided by the CellMeasurer which allows the cell to re-check its size, which will then update the row width calculated by the List component.